### PR TITLE
new hazard API

### PR DESCRIPTION
**Pull request by VALERA771**

### CODE REVIEW

File: Exiled.Events/EventArgs/Player/EnteringEnvironmentalHazardEventArgs.cs [hunk 1]
```diff
@@ -39,6 +40,9 @@ public EnteringEnvironmentalHazardEventArgs(API.Features.Player player, Environm
         /// </summary>
         public EnvironmentalHazard EnvironmentalHazard { get; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Player/EnteringEnvironmentalHazardEventArgs.cs:41]: Obsolete

File: Exiled.Events/EventArgs/Player/ExitingEnvironmentalHazardEventArgs.cs [hunk 2]
```diff
@@ -39,6 +40,9 @@ public ExitingEnvironmentalHazardEventArgs(API.Features.Player player, Environme
         /// </summary>
         public EnvironmentalHazard EnvironmentalHazard { get; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Player/ExitingEnvironmentalHazardEventArgs.cs:41]: Same

File: Exiled.Events/EventArgs/Player/StayingOnEnvironmentalHazardEventArgs.cs [hunk 3]
```diff
@@ -36,5 +37,8 @@ public StayingOnEnvironmentalHazardEventArgs(API.Features.Player player, Environ
         /// Gets the environmental hazard that the player is staying on.
         /// </summary>
         public EnvironmentalHazard EnvironmentalHazard { get; }
```
Comment (louis1706) [inline Exiled.Events/EventArgs/Player/StayingOnEnvironmentalHazardEventArgs.cs:39]: Same

File: Exiled.Events/Patches/Generic/HazardListAdd.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    /// </summary>
+    [HarmonyPatch(typeof(EnvironmentalHazard), nameof(EnvironmentalHazard.Start))]
+    internal class HazardListAdd
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            newInstructions.InsertRange(
+                newInstructions.Count - 1,
```
Comment (louis1706) [inline Exiled.Events/Patches/Generic/HazardListAdd.cs:29]: Would be able to use Postfix but that not a problem we can keep it transpiler

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: Exiled.Events/Patches/Generic/HazardListAdd.cs (hunk 4 above)
Comment (moddedmcplayer) [reply Exiled.Events/Patches/Generic/HazardListAdd.cs:29]: postfix for the win

**Review by louis1706 — CHANGES_REQUESTED** [review]
Hazard is not added in the list when they spawn / do not get Sinkhole in the list (if no one has enter the sinkhole)

if it's get added i would like to see a list of all sinkhole in OnWaitingPlayer (should contain the sinkhole in it)

**Comment by VALERA771** [comment]
Tested by sinkhole hazards

**Review by louis1706 — APPROVED** [review]

**Review by NaoUnderscore — APPROVED** [review]

**Review by Misaka-ZeroTwo — APPROVED** [review]

**Pull request merged**